test1/task1/main.c

```diff
@@ -0,0 +1,124 @@
+﻿#include <stdio.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <limits.h>
+
+#define NUMBER_OF_BITS sizeof(unsigned int) * 8
```

Comment:
```suggestion
#define NUMBER_OF_BITS (sizeof(unsigned int) * 8)
```
Всегда заключайте выражения в скобки, потому что в месте их использования может быть беда с приоритетом операций.